fix: Update dependencies to address security vulnerabilities

gaprl opened the pull request:
## Summary
- Updates multiple dependencies to fix security vulnerabilities reported in Linear tickets
- Ensures compatibility with React 18 and fixes TypeScript compilation issues

## Security Fixes
- **terser DoS vulnerability (RTC-1066, RTC-1059)**: Updated from 4.8.0 to 4.8.1 and 5.13.1 to 5.43.1
- **dompurify XSS/DoS vulnerabilities (RTC-1060, RTC-1061, RTC-1062, RTC-1063)**: Updated from 3.0.0 to 3.1.3
- **remark-html XSS vulnerability (RTC-1064)**: Updated from 12.0.0 to 13.0.2

## Additional Changes
- Updated html-loader from 1.2.1 to 2.1.2 (pulls in fixed terser)
- Updated webpack from 5.72.1 to 5.75.0
- Updated html-webpack-plugin from 5.5.0 to 5.5.4
- Added missing @babel/runtime dependency
- Fixed React 18 compatibility in demo files
- Fixed TypeScript errors for proper compilation

## Test plan
- [x] All tests pass (`yarn test`)
- [x] Build completes successfully (`yarn build`)
- [x] Verified vulnerable package versions are updated
- [x] No runtime errors

gaprl committed:
fix: Update dependencies to address security vulnerabilities

- Update terser from 4.8.0 to 4.8.1 and 5.13.1 to 5.43.1 (fixes RTC-1066, RTC-1059)
- Update dompurify from 3.0.0 to 3.1.3 (fixes RTC-1060, RTC-1061, RTC-1062, RTC-1063)
- Update remark-html from 12.0.0 to 13.0.2 (fixes RTC-1064)
- Update html-loader from 1.2.1 to 2.1.2 to pull in fixed terser
- Update webpack from 5.72.1 to 5.75.0
- Update html-webpack-plugin from 5.5.0 to 5.5.4
- Add missing @babel/runtime dependency
- Fix React 18 compatibility issues in demo files
- Fix TypeScript errors for proper compilation

scttcper commented on package.json:
```diff
@@ -5,11 +5,11 @@
   "author": "Sentry",
   "dependencies": {
     "@types/react": ">=16",
```
probably bump this up to 17 too if we're doing types/react-dom

scttcper commented on package.json:
```diff
@@ -5,11 +5,11 @@
   "dependencies": {
     "@types/react": ">=16",
-    "@types/react-dom": ">=16",
+    "@types/react-dom": "^17.0.0",
```
we can't actually do `^` here since it would force anyone using this package to be on 17.
```suggestion
    "@types/react-dom": ">=17.0.0",
```

gaprl committed:
fix: Address review feedback - move React types to peer/dev dependencies

- Move @types/react and @types/react-dom from dependencies to devDependencies
- Add them to peerDependencies with >=16 to match react/react-dom
- This prevents forcing consumers to use a specific React version

scttcper reviewed (approved):
looks good i think its safe to assume they have it as a peer dependency
